bot/rule.go

```diff
@@ -62,6 +64,45 @@ func (r *rule) Actions() []Action {
 	return r.actions
 }
 
+type rulesGroup struct {
+	key   int
+	rules []Rule
+}
+
+type rulesGroups []rulesGroup
+
+func (r rulesGroups) Len() int {
+	return len(r)
+}
+
+func (r rulesGroups) Less(i, j int) bool {
+	return r[i].key < r[j].key
+}
+
+func (r rulesGroups) Swap(i, j int) {
+	r[i], r[j] = r[j], r[i]
+}
+
+func groupByRuleOrder(rules []Rule) []rulesGroup {
+	groupIndexes := make(map[int]rulesGroup)
+	for _, rule := range rules {
+		key := rule.Order()
+		rules, exists := groupIndexes[key]
```

Comment:
This local var name confused me a bit, also shadows the function's scope - how about ruleGroup?